src/sentry/testutils/cases.py

```diff
@@ -678,6 +680,17 @@ def store_event(self, *args, **kwargs):
             stored_group = stored_event.group
             if stored_group is not None:
                 self.store_group(stored_group)
+            # Verify that events have settled in snuba's storage.
+            # While snuba is synchronous, clickhouse isn't entirely synchronous.
+            attempt = 0
+            while attempt < 10:
+                did_store = eventstore.get_event_by_id(
+                    stored_event.project_id, stored_event.event_id
+                )
+                if did_store:
+                    break
+                attempt += 1
+                time.sleep(0.05)
```

Comment:
`time.sleep(0.05 + attempt*0.05)` for a little backoff? 500ms might be too little total time? I think in terms of fixing flake I'd prefer we just fail the test if `store_event` fails to store the event in a timely matter, because that's the root of the type of flake, and we can look into reducing that time later.

Reply:
Failing the test is an option. I can try that out and see if it triggers often.

Reply:
Yeah, I think if store_event never finishes, it would be a big red flag that we should look into. I'd rather fix it at the source then see it's effects indirectly through different layers of the test.

Reply:
Would this cause problems with the visual snapshots and maybe other tests too?

For example, if a test is storing 2 or more events, the subsequent events that use `before_now` to set the timestamp will be pushed later, so the two events can be further apart in time than expected because of this sleep.

Reply:
If that's the case, we can collect the ids and offer another `wait` method to be called later in the test after all the `store_event`'s have been called, which then would check that they have all been stored at once.

Reply:
> For example, if a test is storing 2 or more events, the subsequent events that use before_now to set the timestamp will be pushed later, so the two events can be further apart in time than expected because of this sleep.

Yes this would push time forward and could result in those events being farther apart than requested. We could not do this and instead have an assertion check that would let test authors wait for a provided event count, or fail. Something like

```python
self.store_event(...)
self.store_event(...)
# Wait for up to 1 second and then fail with an assertion error.
self.wait_for_event_count(2)
```

The 1 second is entirely arbitrary and I don't know yet what a good default value would be.

Reply:
> The 1 second is entirely arbitrary and I don't know yet what a good default value would be. 

I would prefer waiting for the default selenium wait time, as if these events don't get stored then the rest of the test will like break, we can turn it down from there if it causes issues, but that would make it the most resilient to ensure the highest test pass rate when it's not going to fail.

Reply:
The default selenium wait time is ~30 seconds, however this method is also used in API test cases and the total failure scenario duration for those gets really high at 30 seconds per stored event.

Reply:
Oh do we fail fast or collect failures? If we collect them then that's a fair point, I still think 1sec might be too low.

Reply:
Regardless, we can work to make failure times around event storage better. If we have 30 second failures due to event storage on tests, then we should look into it and fix it.

Reply:
We don't fail fast in CI unfortunately.